test/prefix.js

```diff
@@ -111,3 +113,86 @@ prepareAndRunTest('Prefix equal falsy', dir, (t, db, raf) => {
     })
   })
 })
+
+prepareAndRunTest('Prefix equal', dir, (t, db, raf) => {
+  const name = 'Prefix equal'
+
+  const msg1 = { type: 'post', text: 'Testing!' }
+  const msg2 = { type: 'contact', text: 'Testing!' }
+  const msg3 = {
+    type: 'vote',
+    vote: {
+      link: '%wOtfXXopI3mTHL6F7Y3XXNtpxws9mQdaEocNJuKtAZo=.sha256',
+      value: 1,
+      expression: 'Like',
+    },
+  }
+  const msg4 = {
+    type: 'vote',
+    vote: {
+      link: '%wOtfXXopI3mTHL6F7Y3XXNtpxws9mQdaEocNJuKtAZo=.sha256',
+      value: 0,
+      expression: 'UnLike',
+    },
+  }
+  const msg5 = {
+    type: 'vote',
+    vote: {
+      link: '%wOtfXXopI3mTHL6F7Y3XXNtpxws9mQdaEocNJuKtAZo=.sha256',
+      value: 1,
+      expression: 'Like',
+    },
+  }
+
+  let state = validate.initial()
+  state = validate.appendNew(state, null, keys, msg1, Date.now())
+  state = validate.appendNew(state, null, keys, msg2, Date.now() + 1)
+  state = validate.appendNew(state, null, keys, msg3, Date.now() + 2)
+  state = validate.appendNew(state, null, keys, msg4, Date.now() + 3)
+  state = validate.appendNew(state, null, keys, msg5, Date.now() + 4)
+
+  const voteQuery = {
+    type: 'EQUAL',
+    data: {
+      seek: helpers.seekVoteLink,
+      value: Buffer.from(
+        '%wOtfXXopI3mTHL6F7Y3XXNtpxws9mQdaEocNJuKtAZo=.sha256'
+      ),
+      indexType: 'vote',
+      prefix: 32,
+    },
+  }
+
+  addMsg(state.queue[0].value, raf, (err, msg) => {
+    addMsg(state.queue[1].value, raf, (err, msg) => {
+      addMsg(state.queue[2].value, raf, (err, msg) => {
+        db.all(voteQuery, 0, false, false, (err, results) => {
+          t.equal(results.length, 1)
+          t.equal(results[0].value.content.type, 'vote')
+
+          db = jitdb(raf, path.join(dir, 'indexes' + name))
```

Comment:
What's the idea here with reinitializing the jitdb instance?

Reply:
It was easier to trigger the bug this way because of the index trim. Also this is much closer to a realistic test than writing 32k values 😊